Test/Altinn.Broker.Tests/LegacyFileControllerTests.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Specialized;
+using System.Globalization;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Text;
+using System.Text.Json;
+
+using Altinn.Broker.Core.Models;
+using Altinn.Broker.Enums;
+using Altinn.Broker.Models;
+using Altinn.Broker.Tests.Factories;
+using Altinn.Broker.Tests.Helpers;
+
+using Hangfire.Common;
+using Hangfire.States;
+
+using Moq;
+
+using Xunit;
+
+namespace Altinn.Broker.Tests;
+public class LegacyFileControllerTests : IClassFixture<CustomWebApplicationFactory>
+{
+    private readonly CustomWebApplicationFactory _factory;
+    private readonly HttpClient _senderClient;
+    private readonly HttpClient _recipientClient;
+    private readonly JsonSerializerOptions _responseSerializerOptions;
+
+    public LegacyFileControllerTests(CustomWebApplicationFactory factory)
+    {
+        _factory = factory;
+        _senderClient = factory.CreateClient();
+        _senderClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", TestConstants.DUMMY_SENDER_TOKEN);
```

Comment:
Should use DUMMY_LEGACY_TOKEN to make an independent HttpClient here, like _senderClient and _recipientClient. Maybe call it _legacyClient.

Reply:
Agreed; We also only need one _legacyClient as we will use fields in message to specify the correct sender/recipient.